### PR TITLE
fix: mismatching mls thumbprint [WPB-6878]

### PR DESCRIPTION
**Pull request by markusfassbender**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6878" title="WPB-6878" target="_blank">BugWPB-6878</a>  [iOS] Mismatch MLS Thumbprint 
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

See ticket.

### Solutions

- See ticket
- Remove fallback `mlsPublicKeys.ed25519` to avoid mixing values

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

- See ticket

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift [hunk 1]
```diff
@@ -610,7 +610,7 @@ extension ClientListViewController: UserObserving {
 private extension UserClient {
 
     var resolvedMLSThumbprint: String? {
-        e2eIdentityCertificate?.mlsThumbprint ?? mlsPublicKeys.ed25519
+        e2eIdentityCertificate?.mlsThumbprint
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift]: from my understanding this was wrong - does the change affect anything else connected too this view controller?

Comment (KaterinaWire) [reply wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift]: I don't think.
do you think we should keep `resolvedMLSThumbprint`?

Comment (markusfassbender) [reply wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift]: Thanks, done!

**Review by johnxnguyen — APPROVED** [review]

File: wire-ios/Wire-iOS Tests/DeviceManagement/ClientTableViewCellTests.swift [hunk 2]
```diff
@@ -145,10 +145,18 @@ final class ClientTableViewCellTests: BaseSnapshotTestCase {
...
+        .uppercased()
+        .splitStringIntoLines(charactersPerLine: 16)
+
+    private static let mockFingerPrint: String = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijkl"
+        .uppercased()
+        .splitStringIntoLines(charactersPerLine: 16)
+
     static func mock(title: String = "Lorem ipsum",
                      isProteusVerified: Bool = false,
-                     mlsThumbprint: String = .mockFingerPrint,
```
Comment (theswiftway) [inline wire-ios/Wire-iOS Tests/DeviceManagement/ClientTableViewCellTests.swift:150]: **question:**  any reasons why we removed the common mocks and using explicit mocks here ?

File: wire-ios/Wire-iOS Tests/DeviceManagement/DeviceDetailsViewTests.swift [hunk 3]
```diff
@@ -23,13 +23,23 @@ import WireSyncEngineSupport
 import WireRequestStrategySupport
 
 final class DeviceDetailsViewTests: BaseSnapshotTestCase, CoreDataFixtureTestHelper {
+
+    private let mockProteusId: String = "abcdefghijklmnop"
```
Comment (theswiftway) [inline wire-ios/Wire-iOS Tests/DeviceManagement/DeviceDetailsViewTests.swift:27]: **question: **  any reason we are repeating same mocks in every class ?

File: wire-ios/Wire-iOS Tests/DeviceManagement/ClientTableViewCellTests.swift (hunk 2 above)
Comment (markusfassbender) [reply wire-ios/Wire-iOS Tests/DeviceManagement/ClientTableViewCellTests.swift:150]: Yes, the reason is that it is not necessary to add a dependency to a definition in another file. Once you want to move files around (e.g. move class to another module) this becomes very complex and is the main reason why I could not easily split unit tests and snapshot tests in my tries.

File: wire-ios/Wire-iOS Tests/DeviceManagement/DeviceDetailsViewTests.swift (hunk 3 above)
Comment (markusfassbender) [reply wire-ios/Wire-iOS Tests/DeviceManagement/DeviceDetailsViewTests.swift:27]: It's only two places and a simple string - I don't see any problems and the strings can be even different for each test, e.g. you want to have an empty string for one test case.

File: wire-ios/Wire-iOS/Sources/UserInterface/UserProfile/Devices/ProfileDeviceDetailsView.swift [hunk 4]
```diff
@@ -101,12 +110,12 @@ struct ProfileDeviceDetailsView: View {
     var body: some View {
         NavigationView {
             ScrollView {
-                if let thumbprint = viewModel.mlsThumbprint, thumbprint.isNonEmpty {
+                if viewModel.isE2eIdentityEnabled, let thumbprint = viewModel.mlsThumbprint, thumbprint.isNonEmpty {
```
Comment (theswiftway) [inline wire-ios/Wire-iOS/Sources/UserInterface/UserProfile/Devices/ProfileDeviceDetailsView.swift]: **sugesstion:**  For not activated case, mlsview should not be shown as we no longer have mls thumbprint

File: wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift [hunk 5]
```diff
@@ -575,6 +575,23 @@ final class ClientListViewController: UIViewController,
...
+    // MARK: Helpers
+
+    private func makeNotActivatedE2EIdenityCertificate(client: UserClient) -> E2eIdentityCertificate? {
+        guard let mlsResolver = MLSClientResolver().mlsClientId(for: client) else {
+            return nil
+        }
+        return E2eIdentityCertificate(
+            clientId: mlsResolver.rawValue,
+            certificateDetails: "",
+            mlsThumbprint: client.mlsPublicKeys.ed25519 ?? "",
```
Comment (theswiftway) [inline wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift]: **suggestion:** we should remove this mls value as we do not have mlsThumbprint

File: wire-ios/Wire-iOS/Sources/UserInterface/UserProfile/Devices/ProfileDeviceDetailsView.swift (hunk 4 above)
Comment (markusfassbender) [reply wire-ios/Wire-iOS/Sources/UserInterface/UserProfile/Devices/ProfileDeviceDetailsView.swift]: @theswiftway is this what you mean?
f1b693e0c89441ef9311f3c53e71f080a06e8c9c

**Comment by theswiftway** [comment]
Looks good. Could you regenerate snapshot test for Not activated case?

**Review by theswiftway — APPROVED** [review]

**Pull request merged**